crates/ursa-gateway/src/server/route/api/v1/get.rs

```diff
@@ -76,6 +79,37 @@ pub async fn get_block_handler(
 
     debug!("received indexer response for {cid}:\n{indexer_response:?}");
 
+    let resp = match get_provider(&indexer_response) {
+        Some(addrs) => {
+            let mut addr_iter = addrs.iter();
+
+            loop {
+                match addr_iter.next() {
+                    Some(&addr) => {
+                        let server_config = ServerConfig::new(4069, addr.to_string());
```

Comment:
we should read/parse `port` info from the `AddrInfo`.

Reply:
@scott-dn After #170 I will rebase and update this.